File: climb-log-flutter/lib/climasys/api/models/create_competition_request.dart
class CreateCompetitionRequest {
  final List<String> gymNames;
  final String competitionName;
  final DateTime? startDate;
  final DateTime? endDate;
  final List<CompetitionGroupRequest> competitionGroupsRequest;

  CreateCompetitionRequest({
    required this.gymNames,
    required this.competitionName,
    this.startDate,
    this.endDate,
    required this.competitionGroupsRequest,
  });

  Map<String, dynamic> toJson() => {
    'gymNames': gymNames,
    'competitionName': competitionName,
    'startDate': startDate?.toIso8601String(),
    'endDate': endDate?.toIso8601String(),
    'competitionGroupsRequest':
    competitionGroupsRequest.map((e) => e.toJson()).toList(),
  };
}

class CompetitionGroupRequest {
  final String name;
  final int? numberOfClimbsIncluded;

  CompetitionGroupRequest({
    required this.name,
    required this.numberOfClimbsIncluded,
  });

  Map<String, dynamic> toJson() => {
    'name': name,
    'numberOfClimbsIncluded': numberOfClimbsIncluded,
  };
}

File: ClimbLog.Api/ClimbLog.Application/Features/Competitions/Commands/CreateCompetition/CreateCompetitionRequest.cs
using ClimbLog.Application.Features.Competitions.Commands.CreateCompetition.Models;

namespace ClimbLog.Application.Features.Competitions.Commands.CreateCompetition;
public class CreateCompetitionRequest
{
    public required List<string> GymNames { get; set; }
    public required string CompetitionName { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public required List<CompetitionGroupRequest> CompetitionGroupsRequest { get; set; }
}
